packages/store/src/patterns/rankOrder.js

```diff
@@ -66,6 +71,18 @@ const memoOfSorted = new WeakMap();
  */
 const comparatorMirrorImages = new WeakMap();
 
+export const recordParts = record => {
+  assertRecord(record);
+  // TODO Measure which is faster: a reverse sort by sorting and
+  // reversing, or by sorting with an inverse comparison function.
+  // If it makes a significant difference, use the faster one.
+  const names = ownKeys(record).sort().reverse();
+  // @ts-expect-error It thinks name might be a symbol, which it doesn't like.
```

Comment:
We really need to have generic types for `ownKeys` and co.

Reply:
I don't understand. What is generic or generic-able about the type of `ownKeys`?

Reply:
`assertRecord` narrows the type of `record` to `CopyRecord<any>` which means `record: Record<string, any>`, and no keys would be symbols (or number).

The problem is that the type of `ownKeys`: `(target: object) => (string | symbol)[]`. If it was generic, aka `ownKeys<K extends PropertyKey>(target: Record<K, any>) => K[]`, we wouldn't have a need for this `@ts-expect-error`, as the type of keys would flow through.